tests/components/mqtt/test_discovery.py

```diff
@@ -35,16 +40,21 @@
     mock_entity_platform,
     mock_registry,
 )
+from tests.typing import (
+    MqttMockHAClientGenerator,
+    MqttMockPahoClient,
+    WebSocketGenerator,
+)
 
 
 @pytest.fixture
-def device_reg(hass):
+def device_reg(hass: HomeAssistant) -> dr.DeviceRegistry:
```

Comment:
See #87694, maybe the fixtures can be removed in a follow-up PR?

Reply:
Or maybe not... I'm not sure to what extent these two are interchangeable:
- `dev_reg = mock_device_registry(hass)`
- `dev_reg = dr.async_get(hass)`

Reply:
See https://github.com/home-assistant/core/pull/87697

Reply:
Yeah, may be. Lets open a separate PR for that then,